lib/hako/schedulers/ecs_volume_comparator.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'hako/schema'
+
+module Hako
+  module Schedulers
+    class EcsVolumeComparator
+      # @param [Hash] expected_volume
+      def initialize(expected_volume)
+        @expected_volume = expected_volume
+        @schema = volume_schema
+      end
+
+      # @param [Aws::ECS::Types::Volume] actual_volume
+      # @return [Boolean]
+      def different?(actual_volume)
+        !@schema.same?(actual_volume.to_h, @expected_volume)
+      end
+
+      private
+
+      def volume_schema
+        Schema::Structure.new.tap do |struct|
+          struct.member(:docker_volume_configuration, Schema::Nullable.new(docker_volume_configuration_schema))
+          struct.member(:host, Schema::Nullable.new(host_schema))
+          struct.member(:name, Schema::String.new)
+        end
+      end
+
+      def docker_volume_configuration_schema
+        Schema::Structure.new.tap do |struct|
+          struct.member(:autoprovision, Schema::Nullable.new(Schema::Boolean.new))
```

Comment:
Is autoprovision nullable? The documentation says it has default value.
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#volumes

Reply:
```
...
  volumes: {
    foo: {
      docker_volume_configuration: {
      },
    },
  },
...
```

Yes, it seems so. The above leads to the following response:

```
> ecs.describe_task_definition(task_definition: 'bar:1').task_definition.volumes[0].docker_volume_configuration
=> #<struct Aws::ECS::Types::DockerVolumeConfiguration scope="task", autoprovision=nil, driver="local", driver_opts=nil, labels=nil>
```